pkg/virt-handler/vm.go

```diff
@@ -227,8 +218,17 @@ func (d *VirtualMachineController) startDomainNotifyPipe(vmi *v1.VirtualMachineI
 			}
 
 			go func(vmi *v1.VirtualMachineInstance) {
-				log.Log.Object(vmi).Infof("Accepted new notify pipe connection for vmi")
 				defer fd.Close()
+
+				// pipe the VMI domain-notify.sock to the virt-handler domain-notify.sock
+				// so virt-handler receives notifications from the VMI
+				conn, err := net.Dial("unix", filepath.Join(d.virtShareDir, "domain-notify.sock"))
```

Comment:
what if this line success, but after that, and before the go routine is finished, the server is restarted and the socket is closed, are we robust for it ?
due to #3806 it shouldn't happen almost, but still in order to be robust.

Reply:
the copy should fail and break out of the proxy goroutine eventually. in parallel any new connections the client attempts to establish will result in a new proxy goroutine.